### PR TITLE
Final Product

### PR DESCRIPTION
**Pull request by KareemWilson**
## **In this Branch, @Kasaija-Kenneth and I did the following:**

1. Follow the layout of the wireframes provided, and personalize the rest of the design including colors, typographies, spacings, etc.
2. build Login and registration pages with Devise.
3. Build a Food list layout that displays a list of food added by the logged-in user as in the wireframe
4. Build a Recipes list layout that displays a list of recipes created by the logged-in user as in the wireframe.
5. Build a Public recipe list layout to display a list of all public recipes ordered by the newest as in the wireframe.
6. Build a Recipe-details layout
7. Build a General shopping list view
8. Fix the N+1 queries problem
9. Create a navigation menu that allows users to open all of the pages you created.
10. Set up RSpec and Capybara testing libraries.
11.  Create unit tests that test models, controllers, requests, and views

### CODE REVIEW

File: app/views/partials/_navbar.html.erb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      </li>
+      <% if user_signed_in? %>
+      <li class="nav-item d-flex justify-center">
+        <%= link_to 'Sign out', destroy_user_session_path, method: :delete, class: "btn btn-outline-success pb-0 pt-2"%>
+      </li>
+      <% else %>
+        <%= link_to "Sign in", new_user_session_path, class: "btn btn-outline-success mx-2 pb-0 pt-2" %> 
+        <%= link_to "Sign up", new_user_registration_path, class: "btn btn-outline-success mx-2 pt-2" %>
+      <%end%>
+    </ul>
```
Comment (Felix45) [inline app/views/partials/_navbar.html.erb:35]: - [ Required ] Please consider adding a page that shows a listing of all the `public` recipes, as part of the project requirements you are supposed to create a page [http://localhost:3000/public_recipes](http://localhost:3000/public_recipes) that shows all the public recipes as shown in the wireframe below.

## Project Requirements
Screenshot from 2023-02-18 02-26-33

**Review by Felix45 — CHANGES_REQUESTED** [review]
Hi @KareemWilson & @Kasaija-Kenneth , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights :heavy_check_mark: 

- [ ] No linter errors
- [ ] Gitflow is used correctly :white_check_mark: 
- [ ] Good UI design :white_check_mark: 
- [ ] Descriptive pull request title and summary :white_check_mark: 


## Required Changes ♻️

- [ Required ] Please customize the [YOUR NAME] section on the `Mit.md` file in your project directory, this will show anyone who would like to use your code who `Licensed` the project.

- [Required] Please remove the default boilerplate instructions on your readme file, they make your repository look unprofessional.

Screenshot from 2023-02-18 02-48-40

Screenshot from 2023-02-18 02-48-22

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/partials/_navbar.html.erb (hunk 1 above)
Comment (KareemWilson) [reply app/views/partials/_navbar.html.erb]: Thank you so much, I did all the required changes.

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @KareemWilson & @Kasaija-Kenneth ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kasaija-Kenneth** [comment]
We are Much obliged @1GORDON and @Felix45

**Pull request merged**